misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -253,6 +253,9 @@ internal class RealTransacter private constructor(
     if (config.type.isVitess) {
       return sessionFactory
     }
+    if (config.type == DataSourceType.COCKROACHDB || config.type == DataSourceType.TIDB) {
```

Comment:
nit: can it be collapsed into one single conditional for vitess, crdb and tidb?

Reply:
For TiDB, we should probably directly use follower reads and don't even call this function? Something like:

```
  override fun <T> replicaRead(block: (session: Session) -> T): T {
    check(!inTransaction) { "Can't do replica reads inside a transaction" }
    try {
      session.useConnection { connection ->
        connection.createStatement().use { s ->
          s.execute("set @@tidb_replica_read = 'follower'")
        }
      }
      block(session)
    } finally {
      session.useConnection { connection ->
        connection.createStatement().use { s ->
          s.execute("set @@tidb_replica_read = ''")
        }
      }
    }
  }
```

Reply:
OH!! I was looking for something like that but couldn't find anything googling "tidb replica reads"